File: Java Labs/Patterns On Java/src/com/pattern/proxy/ClothingImageProxy.java
package com.pattern.proxy;

public class ClothingImageProxy implements ClothingImage {
    private RealClothingImage realImage;
    private String imageUrl;

    public ClothingImageProxy(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    @Override
    public void display() {
        if (realImage == null) {
            realImage = new RealClothingImage(imageUrl);
        }
        realImage.display();
    }
}

File: Python Labs/com/pattern/proxy/ClothingImageProxy.py
from com.pattern.proxy.ClothingImage import ClothingImage
from com.pattern.proxy.RealClothingImage import RealClothingImage


class ClothingImageProxy(ClothingImage):
    def __init__(self, image_url):
        self.image_url = image_url
        self.real_image = None

    def display(self):
        if self.real_image is None:
            self.real_image = RealClothingImage(self.image_url)
        self.real_image.display()